docs/src/design/stack/op_constraints.md

```diff
@@ -45,20 +45,22 @@ $$
 
 As can be seen from above, the degree for both of these flags is $7$. Since degree of constraints in Miden VM can go up to $9$, this means that operation-specific constraints cannot exceed degree $2$. However, there are some operations which require constraints of higher degree (e.g., $3$ or even $5$). To support such constraints, we adopt the following scheme.
 
-We organize the operations into $3$ groups as shown below and also introduce an extra register for degree reduction:
+We organize the operations into $4$ groups as shown below and also introduce two extra registers $e_0$ and $e_1$ for degree reduction:
 
-| $b_6$ | $b_5$ | $b_4$ | $b_3$ | $b_2$  | $b_1$ | $b_0$ | extra  | # of ops | degree  |
-| :---: | :---: | :---: | :---: | :----: | :---: | :---: | :----: |:-------: | :-----: |
-| 0     |  x    | x     | x     | x      | x     | x     |  0     | 64       |  7      |
-| 1     |  0    | x     | x     | x      | x     | -     |  0     | 16       |  6      |
-| 1     |  1    | x     | x     | x      | -     | -     |  1     | 8        |  4      |
+| $b_6$ | $b_5$ | $b_4$ | $b_3$ | $b_2$  | $b_1$ | $b_0$ | $e_0$ | $e_1$  |# of ops | degree  |
+| :---: | :---: | :---: | :---: | :----: | :---: | :---: | :---: | :----: | :-----: | :-----: |
+| 0     |  x    | x     | x     | x      | x     | x     | 0     | 0      | 64      | 7       |
+| 1     |  0    | 0     | x     | x      | x     | -     | 0     | 0      | 8       | 6       |
+| 1     |  0    | 1     | x     | x      | x     | x     | 1     | 0      | 16      | 5       |
+| 1     |  1    | x     | x     | x      | -     | -     | 0     | 1      | 8       | 4       |
 
 In the above:
 * Operation flags for operations in the first group (with prefix `0`), are computed using all $7$ op bits, and thus their degree is $7$.
-* Operation flags for operations in the second group (with prefix `10`), are computed using only the first $6$ op bits, and thus, their degree is $6$.
-* Operation flags for operations in the third group (with prefix `11`), are computed using only the first $5$ op bits, and we also use the extra register (which is set to $b_6 \cdot b_5$) to reduce the degree by $1$. Thus, the degree of op flags in this group is $4$.
+* Operation flags for operations in the second group (with prefix `100`), are computed using only the first $6$ op bits, and thus their degree is $6$.
+* Operation flags for operations in the third group (with prefix `101`), are computed using all $7$ op bits. We use the extra register $e_0$ (which is set to $b_6 \cdot (1-b_5) \cdot b_4$) to reduce the degree by $2$. Thus, the degree of op flags in this group is $5$.
+* Operation flags for operations in the fourth group (with prefix `11`), are computed using only the first $5$ op bits. We use the extra register $e_1$ (which is set to $b_6 \cdot b_5$) to reduce the degree by $1$. Thus, the degree of op flags in this group is $4$.
```

Comment:
In #884 , the expressions for `e0` and `e1` are the other way around but I think that is a mistake there. This PR has things right, as far as I can see.

Reply:
ah I see - in the picture it's shown the way I did it, but in the written description it's flipped. You can tell I'm a visual person :)